### PR TITLE
Add 2D Langmuir Test with MR

### PR DESCRIPTION
**Pull request by EZoni**
This adds a first 2D Langmuir test with mesh refinement ratio 4 and FDTD (CKC) solver. The Python analysis comparing the output fields with their theoretical predictions passes.

I believe that increasing the refinement ratio will cause problems, which I will report in a separate issue once the test is merged and it's easy to tune its input parameters.

### CODE REVIEW

File: Examples/Tests/Langmuir/analysis_langmuir_multi_2d.py [hunk 1]
```diff
@@ -35,88 +42,116 @@
...
+dx = Lx / Nx
+dz = Lz / Nz
 
-def get_contribution( is_cos, k ):
-    du = (xmax-xmin)/Nx
-    u = xmin + du*( 0.5 + np.arange(Nx) )
-    if is_cos == 1:
-        return( np.cos(k*u) )
-    else:
-        return( np.sin(k*u) )
```
Comment (EZoni) [inline Examples/Tests/Langmuir/analysis_langmuir_multi_2d.py]: This function, now removed, was using always the parameters along x (`xmin`, `xmax`, `Nx`), for both the x and z contributions, which would be potentially wrong if we were to set up some of the parameters in a way that is not symmetric.

File: Examples/Tests/Langmuir/analysis_langmuir_multi_2d.py [hunk 2]
```diff
@@ -35,88 +42,116 @@
...
-print("error_rel    : " + str(error_rel))
-print("tolerance_rel: " + str(tolerance_rel))
-
-assert( error_rel < tolerance_rel )
+    E_th = get_theoretical_field(field, t)
+    max_error = abs(E_sim - E_th).max() / abs(E_th).max()
+    print('{:s}: max error = {}'.format(field, max_error))
+    error_rel = max(error_rel, max_error)
+
+tolerance = 0.05
```
Comment (EZoni) [inline Examples/Tests/Langmuir/analysis_langmuir_multi_2d.py]: Note that the tolerance has not been modified.

File: Examples/Tests/Langmuir/analysis_langmuir_multi_2d.py [hunk 3]
```diff
@@ -35,88 +42,116 @@
...
+ax2.set_title('Ez (theory)')
+cb2.ax.tick_params()
+cb2.ax.yaxis.get_offset_text().set()
+cb2.ax.yaxis.offsetText.set_ha('center')
+cb2.ax.yaxis.offsetText.set_va('bottom')
+cb2.formatter.set_powerlimits((0,0))
+cb2.update_ticks()
+# Save figure
+fig.tight_layout()
+fig.savefig('analysis_langmuir_multi_2d.png', dpi=200)
```
Comment (EZoni) [inline Examples/Tests/Langmuir/analysis_langmuir_multi_2d.py:140]: This new code should produce better plots. Below is a comparison of the plots produced by the previous analysis script and the current one.

Before this PR:
<p align="center">
    
</p>

After this PR:
<p align="center">
    
</p>

File: Examples/Tests/Langmuir/analysis_langmuir_multi_2d.py [hunk 4]
```diff
@@ -2,31 +2,38 @@
...
+# This is a script that analyzes the simulation results of the following CI tests:
+# - Langmuir_multi_2d_nodal
+# - Langmuir_multi_2d_psatd
+# - Langmuir_multi_2d_psatd_nodal
+# - Langmuir_multi_2d_psatd_momentum_conserving
+# - Langmuir_multi_2d_psatd_current_correction
+# - Langmuir_multi_2d_psatd_current_correction_nodal
+# - Langmuir_multi_2d_psatd_Vay_deposition
+# - Langmuir_multi_2d_psatd_Vay_deposition_nodal
+# - Langmuir_multi_2d_MR
```
Comment (EZoni) [inline Examples/Tests/Langmuir/analysis_langmuir_multi_2d.py]: This should be a comprehensive list of CI tests for which this analysis script is currently used.

File: Regression/WarpX-tests.ini [hunk 5]
```diff
@@ -473,6 +473,25 @@ analysisRoutine = Examples/Tests/Langmuir/analysis_langmuir_multi_2d.py
 analysisOutputImage = langmuir_multi_2d_analysis.png
 tolerance = 1.e-14
 
+[Langmuir_multi_2d_MR]
+buildDir = .
+inputFile = Examples/Tests/Langmuir/inputs_2d_multi_rt
+runtime_params = algo.maxwell_solver = ckc  warpx.use_filter = 1  amr.max_level = 1  amr.ref_ratio = 4  warpx.fine_tag_lo = -10.e-6 -10.e-6  warpx.fine_tag_hi = 10.e-6 10.e-6  diag1.electrons.variables = w ux uy uz  diag1.positrons.variables = w ux uy uz
```
Comment (EZoni) [inline Regression/WarpX-tests.ini:479]: Here is where we set the input parameters relevant for mesh refinement: `amr.max_level`, `amr.ref_ratio`, `warpx.fine_tag_lo` and `warpx.fine_tag_hi`.

File: Examples/Tests/Langmuir/analysis_langmuir_multi_2d.py [hunk 6]
```diff
@@ -35,88 +41,109 @@
...
+cb1 = fig.colorbar(im1, cax = cax1)
+set_axes_and_colorbar(ax1, cb1, 'Ez (sim)')
+# Second plot
+cax2 = make_axes_locatable(ax2).append_axes('right', size='5%', pad='5%')
+im2 = ax2.imshow(E_th, origin = 'lower', extent = edge[0], vmin = vmin, vmax = vmax)
+cb2 = fig.colorbar(im2, cax = cax2)
+set_axes_and_colorbar(ax2, cb2, 'Ez (theory)')
+# Save figure
+fig.tight_layout()
+fig.savefig('analysis_langmuir_multi_2d.png', dpi=200)
```
Comment (EZoni) [inline Examples/Tests/Langmuir/analysis_langmuir_multi_2d.py]: This new code should produce better plots. Below is a comparison of the plots produced by the previous analysis script and the current one, for one sample test (`Langmuir_multi_2d_nodal`).

Before this PR:
<p align="center">
    
</p>

After this PR:
<p align="center">
    
</p>

**Review by RemiLehe — APPROVED** [review]

**Pull request merged**